src/cmd/cli/command/commands.go

```diff
@@ -825,30 +824,25 @@ var composeUpCmd = &cobra.Command{
 			return nil
 		}
 
-		tailCtx, cancelTail := context.WithCancel(ctx)
+		tailCtx, cancelTail := contextWithServiceStatus(ctx, cli.ServiceStarted, serviceInfos)
+		defer cancelTail()
```

Comment:
how can this happen? startTailing blocks? You do `<-Done()` so this cancel is just for cleanup, in which case the name is wrong.